source/game/field/obj/ObjectShip64.cc

```diff
@@ -0,0 +1,97 @@
+#include "ObjectShip64.hh"
+
+namespace Field {
+
+/// @addr{0x80765C94}
+ObjectShip64::ObjectShip64(const System::MapdataGeoObj &params) : ObjectCollidable(params) {}
+
+/// @addr{0x80765DB0}
+ObjectShip64::~ObjectShip64() = default;
+
+/// @addr{0x80765E30}
+void ObjectShip64::init() {
+    m_railInterpolator->init(0, 0);
+
+    m_tangent = m_railInterpolator->curTangentDir();
+    m_railInterpolator->setPerPointVelocities(true);
+
+    calc();
+    calcModel();
+}
+
+/// @addr{0x80766144}
+void ObjectShip64::calc() {
+    m_railInterpolator->calc();
+
+    setPos(m_railInterpolator->curPos());
+
+    m_flags.setBit(eFlags::Position);
+    m_tangent = Interpolate(0.2, m_tangent, m_railInterpolator->curTangentDir());
+    m_tangent.normalise();
+
+    setMatrixFromOrthonormalBasisAndPos(m_tangent);
+}
+
+/// @addr{0x80766BC0}
+const char *ObjectShip64::getKclName() const {
+    return "DKShip64";
+}
+
+/// @addr{0x80766BB4}
+const char *ObjectShip64::getName() const {
+    return "DKShip64";
+}
+
+/// @addr{0x80766BB4}
+const char *ObjectShip64::getResources() const {
+    return "DKShip64";
+}
+
+/// @addr{0x80766CA4}
+u32 ObjectShip64::loadFlags() const {
+    return 1;
+}
```

Comment:
Can you move all these to the header? We typically move any super tiny/simple functions in there